x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/proto/content/GeneratorUtils.java

```diff
@@ -0,0 +1,254 @@
+            }
+        };
+    }
+
+    private static Writer valuesWriter(Writer writer) {
+        return (g, v) -> {
+            if (v == null) {
+                g.writeNull();
+            } else {
+                g.writeStartArray();
+                writer.write(g, v);
+                g.writeEndArray();
+            }
+        };
+    }
+
+    private static final Map<Class<?>, Writer> WRITERS = new HashMap<>();
+
+    static {
+        WRITERS.put(Boolean.class, valueWriter((p, v) -> p.writeBoolean((Boolean) v)));
+        WRITERS.put(boolean[].class, valuesWriter((p, v) -> {
+            for (boolean b : (boolean[]) v) {
+                p.writeBoolean(b);
+            }
+        }));
+        WRITERS.put(Byte.class, valueWriter((p, v) -> p.writeNumber((Byte) v)));
+        WRITERS.put(byte[].class, valuesWriter((p, v) -> {
+            for (byte b : (byte[]) v) {
+                p.writeNumber(b);
+            }
+        }));
+        WRITERS.put(Date.class, valueWriter((p, v) -> p.writeNumber(((Date) v).getTime())));
+        WRITERS.put(Double.class, valueWriter((p, v) -> p.writeNumber((Double) v)));
+        WRITERS.put(double[].class, valuesWriter((p, v) -> {
+            for (double d : (double[]) v) {
+                p.writeNumber(d);
+            }
+        }));
+        WRITERS.put(Float.class, valueWriter((p, v) -> p.writeNumber((Float) v)));
+        WRITERS.put(float[].class, valuesWriter((p, v) -> {
+            for (float f : (float[]) v) {
+                p.writeNumber(f);
+            }
+        }));
+        WRITERS.put(Integer.class, valueWriter((p, v) -> p.writeNumber((Integer) v)));
+        WRITERS.put(int[].class, valuesWriter((p, v) -> {
+            for (int i : (int[]) v) {
+                p.writeNumber(i);
+            }
+        }));
+        WRITERS.put(Long.class, valueWriter((p, v) -> p.writeNumber((Long) v)));
+        WRITERS.put(long[].class, valuesWriter((p, v) -> {
+            for (long l : (long[]) v) {
+                p.writeNumber(l);
+            }
+        }));
+        WRITERS.put(Short.class, valueWriter((p, v) -> p.writeNumber((Short) v)));
+        WRITERS.put(short[].class, valuesWriter((p, v) -> {
+            for (short s : (short[]) v) {
+                p.writeNumber(s);
+            }
+        }));
+        WRITERS.put(String.class, valueWriter((p, v) -> p.writeString((String) v)));
+        WRITERS.put(String[].class, valuesWriter((p, v) -> {
+            for (String s : (String[]) v) {
+                p.writeString(s);
+            }
+        }));
+        WRITERS.put(Locale.class, valueWriter((p, v) -> p.writeString(v.toString())));
+        WRITERS.put(Class.class, valueWriter((p, v) -> p.writeString(v.toString())));
+        WRITERS.put(ZonedDateTime.class, valueWriter((p, v) -> p.writeString(StringUtils.toString(v))));
+        // writers.put(Calendar.class, XContentBuilder::timeValue);
+        // writers.put(GregorianCalendar.class, XContentBuilder::timeValue);
```

Comment:
leftover?

Reply:
I've left them thinking they might be needed - considering the amount of testing done, I've got my answer.